src/api/coupons/shared/couponsDDBBFunctions.ts

```diff
@@ -1,16 +1,13 @@
-import type {CouponType} from 'src/@types/coupon';
 import type {Categories, DiscountItem, Discounts} from 'src/@types/discount';
 import {STORAGE} from 'src/config';
-import useLocalStorage from 'src/hooks/useLocalStorage';
 
-export function retrieveCouponFromFakeDDBB(id: string): DiscountItem | undefined {
-  const {getStorageItem} = useLocalStorage();
-  const discounts: Discounts = getStorageItem(STORAGE.discounts) as Discounts;
+export function retrieveCouponFromFakeDDBB(id: string): DiscountItem {
+  const discounts: Discounts | undefined = getFromDDBB<Discounts>(STORAGE.discounts);
+  let coupon: DiscountItem | undefined;
   if (!discounts) {
```

Comment:
En este caso no sería necesario tb comprobar que no sea un array vacío?
```suggestion
  if (!discounts || !discounts.length) {
```